sync2/migration.go

```diff
@@ -0,0 +1,264 @@
+package sync2
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"github.com/jmoiron/sqlx"
+	"github.com/matrix-org/sliding-sync/sqlutil"
+)
+
+func MigrateDeviceIDs(db *sqlx.DB, secret string, whoamiClient Client, commit bool) error {
```

Comment:
This isn't called at present. I'll need to write some logic that tries to actually do the migration on startup.